core/node/genesis/src/utils.rs

```diff
@@ -41,78 +40,60 @@ pub(super) async fn add_eth_token(transaction: &mut Connection<'_, Core>) -> any
             let known_code_key = get_known_code_key(&hash);
             let marked_known_value = H256::from_low_u64_be(1u64);
-            (
-                H256::default(),
-                vec![StorageLog::new_write_log(
-                    known_code_key,
-                    marked_known_value,
-                )],
-            )
+
+            StorageLog::new_write_log(known_code_key, marked_known_value)
         })
-        .dedup_by(|a, b| a.1 == b.1)
+        .dedup_by(|a, b| a == b)
         .collect();
 
     let storage_logs: Vec<_> = system_contracts
         .iter()
         .map(|contract| {
             let hash = hash_bytecode(&contract.bytecode);
             let code_key = get_code_key(contract.account_id.address());
-            (
-                H256::default(),
-                vec![StorageLog::new_write_log(code_key, hash)],
-            )
+            StorageLog::new_write_log(code_key, hash)
         })
-        .chain(Some(system_context_init_logs))
+        .chain(system_context_init_logs)
         .chain(known_code_storage_logs)
         .collect();
 
     storage_logs
 }
 
-pub(super) fn get_deduped_log_queries(storage_logs: &[(H256, Vec<StorageLog>)]) -> Vec<LogQuery> {
+pub(super) fn get_deduped_log_queries(storage_logs: &[StorageLog]) -> Vec<LogQuery> {
     // we don't produce proof for the genesis block,
     // but we still need to populate the table
     // to have the correct initial state of the merkle tree
     let log_queries: Vec<MultiVmLogQuery> = storage_logs
         .iter()
-        .enumerate()
-        .flat_map(|(tx_index, (_, storage_logs))| {
-            storage_logs
-                .iter()
-                .enumerate()
-                .map(move |(log_index, storage_log)| {
-                    MultiVmLogQuery {
-                        // Monotonically increasing Timestamp. Normally it's generated by the VM, but we don't have a VM in the genesis block.
-                        timestamp: MultiVMTimestamp(((tx_index << 16) + log_index) as u32),
-                        tx_number_in_block: tx_index as u16,
-                        aux_byte: 0,
-                        shard_id: 0,
-                        address: *storage_log.key.address(),
-                        key: h256_to_u256(*storage_log.key.key()),
-                        read_value: h256_to_u256(H256::zero()),
-                        written_value: h256_to_u256(storage_log.value),
-                        rw_flag: storage_log.kind == StorageLogKind::Write,
-                        rollback: false,
-                        is_service: false,
-                    }
-                })
-                .collect::<Vec<_>>()
+        .map(move |storage_log| {
+            MultiVmLogQuery {
+                // Monotonically increasing Timestamp. Normally it's generated by the VM, but we don't have a VM in the genesis block.
+                timestamp: MultiVMTimestamp(0),
+                tx_number_in_block: 0,
```

Comment:
Nit: let's change the comment explaining that it's fine to set 0 for timestamp and tx_number_in_block here